### DISCUSSION TITLE
timing for 3.11 compatibility?

### DISCUSSION THREAD

**Discussion by cwe6279** [aurelio-labs/semantic-router#132] [General]
would love to give this a try but recently had to update langchain, openai pinecone-client and running 3.11 
At the moment it gives a pydantic compatibility issue in 3.11 on debian:
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
canopy-sdk 0.6.0 requires pydantic<2.0.0,>=1.10.7, but you have pydantic 2.6.0 which is incompatible.
canopy-sdk 0.6.0 requires tiktoken<0.4.0,>=0.3.3, but you have tiktoken 0.5.2 which is incompatible.
```

I cant downgrade due to other openai and langchain dependencies

**Comment by jamescalam** [comment]
seems like canopy-sdk is causing the conflicts, could I suggest installing canopy in a different environment and possibly first confirming if it is in fact canopy causing the conflict with a quick `pip uninstall` followed by `pip install`? Afaik latest langchain versions also require pydantic v2
